monix-execution/jvm/src/main/java/monix/execution/workstealing/LocalQueueConstants.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2014-2021 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This code is a direct copy of the Cats Effect 3 source code, donated by the
+ * code authors and maintainers of the original code.
+ *
+ * The original Cats Effect 3 source code can be found at
+ * `https://github.com/typelevel/cats-effect/blob/27bb3fdffe86d299b8966eb352fa1b2fd8125395/core/jvm/src/main/java/cats/effect/unsafe/LocalQueueConstants.java`.
+ */
+
+package monix.execution.workstealing;
+
+/**
+ * Contains static constants used throughout the implementation of the
+ * [[monix.execution.workstealing.LocalQueue]].
+ */
+class LocalQueueConstants {
```

Comment:
Out of curiosity, why is this a Java file? Just curious, the bulk of the implementation seems to be in Scala.

Reply:
Optimization really. Scala `val`s are fields of object instances, and these are static constants. Loading static constants is faster than loading object fields. 😄

Reply:
True. Although if you add them as `final val` in Scala, they'll be inlined AFAIK.

Reply:
I will definitely test that. Thanks.

Reply:
@alexandru 

Scala code:
```scala
object Main {
  final val n: Byte = 1
}
```

Compiles to the following:
```
public final class Main$ {
  public static final Main$ MODULE$;

  private static final byte n;

  public static {};
    Code:
       0: new           #2                  // class Main$
       3: dup
       4: invokespecial #14                 // Method "<init>":()V
       7: putstatic     #16                 // Field MODULE$:LMain$;
      10: iconst_1
      11: putstatic     #18                 // Field n:B
      14: return

  public final byte n();
    Code:
       0: getstatic     #18                 // Field n:B
       3: ireturn

  private Main$();
    Code:
       0: aload_0
       1: invokespecial #21                 // Method java/lang/Object."<init>":()V
       4: return
}
```

You're right, the field does indeed get inlined to a constant, and the virtual call `n()` can be inlined too. Let me know how you want to proceed.

Reply:
Well, if that satisfies your needs, I'd rather have the Scala file. We have some Java files too, I believe, for the atomic stuff, but this case seems simple enough.